argh_derive/src/lib.rs

```diff
@@ -302,7 +303,8 @@ fn impl_from_args_struct(
 
 fn impl_from_args_struct_from_args<'a>(
     errors: &Errors,
-    type_attrs: &TypeAttrs,
+    #[cfg(feature = "help")] type_attrs: &TypeAttrs,
```

Comment:
I think this would be cleaner if you used `if cfg!(feature = help` to gate the places where the feature is enabled. You'd get fewer unused variables since the compiler would type-check both branches regardless, and the code would flow better.